pkg/serdes/serdes.go

```diff
@@ -22,6 +22,7 @@ type SerializationProvider interface {
 	LoadSchema(string, map[string]string) error
 	Serialize(string) ([]byte, error)
 	GetSchemaName() string
+	IsSchemaBased() bool
```

Comment:
Is there any way to infer this from the other functions, for instance, `GetSchemaName() != ""`?